frontend/public/components/utils/dropdown.jsx

```diff
@@ -221,38 +238,57 @@ export class Dropdown extends DropdownMixin {
       return;
     }
 
-    const { items, keyboardHoverKey } = this.state;
+    const { items, keyboardHoverKey, selectedKey } = this.state;
 
     if (key === 'Enter') {
       if (!keyboardHoverKey) {
         this.hide(e);
         return;
       }
-      this.onClick_(keyboardHoverKey, e);
+
+      if (this.state.active) {
+        this.onClick(keyboardHoverKey, e);
+        return;
+      }
+
+      this.toggle(e);
       return;
     }
 
     const keys = Object.keys(items);
-
     let index = _.indexOf(keys, keyboardHoverKey);
 
+    const { current } = this.dropdownList;
+
+    // so namespace and Events resource selector advance correctly
+    if (selectedKey && index === -1) {
+      const correctIndex = _.indexOf(keys, selectedKey);
+      index = correctIndex;
+    }
+
     if (key === 'ArrowDown') {
       index += 1;
+      if (index > 0) {
+        current.scrollTop += 20;
```

Comment:
Instead of scrolling on every keystroke, should this use something like `scrollIntoView`?

https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView